### PR TITLE
Updated GhostHUD

### PR DESCRIPTION
**Pull request by markusmarkusz**
This is untested.
So please test and tell me if it works. Then I have to fix it.

### CODE REVIEW

File: lua/cl_specdm_hud.lua [hunk 1]
```diff
@@ -78,77 +80,78 @@ local function PaintBar(x, y, w, h, colors, value)
...
-			end
-			if LocalPlayer():GetActiveWeapon().Primary then
-				local ammo_clip, ammo_max, ammo_inv = GetAmmo(LocalPlayer())
-				if ammo_clip != -1 then
-					local ammo_y = health_y + bar_height + margin
-					PaintBar(x+margin, ammo_y, bar_width, bar_height, ammo_colors, ammo_clip/ammo_max)
-					local text = string.format("%i + %02i", ammo_clip, ammo_inv)
-					ShadowedText(text, "HealthAmmo", bar_width, ammo_y, COLOR_WHITE, TEXT_ALIGN_RIGHT, TEXT_ALIGN_RIGHT)
-				end
+local GetLang = LANG.GetUnsafeLanguageTable
```
Comment (MinIsMin) [inline lua/cl_specdm_hud.lua]: Sine LANG is not in the Initialize hook, it throws an error when joining

File: lua/cl_specdm_hud.lua [hunk 2]
```diff
@@ -78,77 +80,78 @@ local function PaintBar(x, y, w, h, colors, value)
...
-	local ttt_health_label = GetConVar("ttt_health_label")
-
-	local margin = 10
-	local old_DrawHUD = GAMEMODE.HUDPaint
-	function GAMEMODE:HUDPaint()
-		if LocalPlayer():IsGhost() then
-			self:HUDDrawTargetID()
-			MSTACK:Draw(LocalPlayer())		
-			TBHUD:Draw(LocalPlayer())
-			WSWITCH:Draw(LocalPlayer())
```
Comment (MinIsMin) [inline lua/cl_specdm_hud.lua]: Better not remove WSWITCH:Draw(LocalPlayer()).

**Comment by MinIsMin** [comment]
The "In progress" tile on the left is not hidden.

File: lua/cl_specdm_hud.lua [hunk 3]
```diff
@@ -78,77 +80,78 @@ local function PaintBar(x, y, w, h, colors, value)
...
-		return old_DrawHUD(self)
 	end
-		
-end)
\ No newline at end of file
+end)
+
+hook.Add("HUDShouldDraw", "GhostHUD", function(n)
+	local p = LocalPlayer()
+	if name == "TTTSpecHUD" and p:IsPlayer() and p:IsGhost() then return false end
```
Comment (Tommy228) [inline lua/cl_specdm_hud.lua]: Wrong variable name here.

Comment (MinIsMin) [inline lua/cl_specdm_hud.lua]: Is already fixed in his next commit: https://github.com/Tommy228/TTT_Spectator_Deathmatch/pull/44/commits/045fc17f2c1f7f8bc1ef40be48990af7a241a78d

**Comment by MinIsMin** [comment]
This is overwriting custom Huds. Maybe we add a setting like CustomScoreboard...

**Comment by markusmarkusz** [comment]
Not needed. HUD Creators can simply remove these two hooks.

**Pull request merged**